packages/zowe-explorer/src/dataset/DatasetTree.ts

```diff
@@ -473,6 +472,10 @@ export class DatasetTree extends ZoweTreeProvider implements IZoweTree<IZoweData
      */
     public deleteSession(node: IZoweDatasetTreeNode) {
         this.mSessionNodes = this.mSessionNodes.filter((tempNode) => tempNode.label.trim() !== node.label.trim());
+        // if we only have favorites node, remove it so that viewsWelcome button shows up
+        if (this.mSessionNodes.length === 1) {
```

Comment:
Soo, why we are deleting the 'Favorite' session node? 

I mean, in case I have have something in 'Favorites' - I don't want to use the proposed welcome screen, but it still appears.

Maybe we can keep 'Favorite' node separately inside tree provider,  instead of mixing all nodes in `mSessionNodes` variable? In that case, we would have to verify, that 'Favorites' is empty and show the welcome view, that we designed. Does it make sense?

Reply:
Without profiles, there is no way to connect to a z/OS LPAR. So there is no way to make use of the favourites. So I believe it is correct that we show the welcome view.

I like the idea of separating the nodes rather than having magic numbers. But I wanted to limit the scope of changes just to adding the new welcome view.

Reply:
Hmm, but I have my profiles in the file system, they also available via 'Favorites' and I can expand them and access LPAR, even if they were removed from the tree (except 'Favorites' node).

Will we remove this ability?

Reply:
Interesting, I just tried it out and you are right @nickImbirev. Thanks for pointing that out! I guess I do not understand the favourites section. It looks like it duplicates the profile tree, only being less powerful (individual data sets instead of filters).

Reply:
Ok, I think I get it now. The search filter will not survive vscode restarts, but the favourites do. Thank you @jellypuno. 

So I believe we should not display the welcome view if favourites exist as you suggested @nickImbirev.